### ISSUE TITLE
Error when a result is not used

### ISSUE THREAD

**Issue by martinheidegger** [golang/lint#287]
We had recently to work through our go code looking for all the errors that we ignored.

```go
io.Copy(reader, file)
```

for example has a return type of `(written int64, err error)` but we subconsciously ignored the error without using `_` in a verbose manner. It would be good if we could have an error message that alerts us on that mishap.

**Comment by ernsheong** [comment]
Yes, I second this. A lot of gotchas in go seem to happen because errors were not handled, and the linter isn't complaining about this.

**Comment by dsnet** [comment]
Thanks for the report, but this is matter of correctness and not a matter of style. Thus, this does not belong in lint. Also there other tools that do this analysis: https://github.com/kisielk/errcheck

**Comment by martinheidegger** [comment]
I am sorry but both writing styles are technically correct. The one I am mentioning here is about an optional error that is invisible which is arguably a bad coding style, am I wrong here?!

**Comment by dsnet** [comment]
There are many legitimate cases where the error can be safely ignored. So flagging this as a matter of style would lead to too many false positives. A separate tool `errcheck` is useful when someone wants to actually vet every ignored error.
